aocr/util/data_gen.py

```diff
@@ -67,6 +68,19 @@ def __init__(self,
         dataset = dataset.map(self._parse_record)
         dataset = dataset.shuffle(buffer_size=10000)
         self.dataset = dataset.repeat(self.epochs)
+        self.train_special_chars = train_special_chars
+        
+        # set CHARMAP to be special characters only and choose a character from the full ascii list to
+        # used for masking out labels of characters not wanted for training.
+        if self.train_special_chars:
+            train_special_chars = [char for char in train_special_chars]
+            if '*' not in self.train_special_chars:
+                self.mask_char = '*'
+            else:
+                self.set_full_ascii_charmap()
+                self.mask_char = random.choice(list(set(self.CHARMAP) - set(train_special_chars)))
```

Comment:
Super minor, but a simpler operation might be
```suggestion
                self.mask_char = next(c for c in self.CHARMAP if c not in train_special_chars)
```
This just uses the first character in `CHARMAP` that isn't in the training characters.